### PR TITLE
Prevent duplicate degree subject

### PR DESCRIPTION
**Pull request by martyn-w**
This change improves the synchronisation of qualification records:

* adds checks to ensure a qualification record is not re-created if it already exists
* creates qualification records immediately, rather than via a queued job

### CODE REVIEW

File: GetIntoTeachingApi/Models/SchoolsExperience/SchoolsExperienceSignUp.cs [hunk 1]
```diff
@@ -132,5 +150,25 @@ private void AcceptPrivacyPolicy(Candidate candidate)
...
+                    DegreeStatusId = DegreeStatusId,
+                    DegreeSubject = DegreeSubject,
+                    TypeId = DegreeTypeId ?? (int)CandidateQualification.DegreeType.Degree
+                });
+            }
+        }
+        
+        private bool ContainsQualification()
+        {
+            return UkDegreeGradeId != null || DegreeStatusId != null || DegreeSubject != null || DegreeTypeId != null;
```
Comment (asatwal) [inline GetIntoTeachingApi/Models/SchoolsExperience/SchoolsExperienceSignUp.cs:171]: Are any of these fields mandatory for a Degree to be valid ? Vould missing fiedls here cause problems with the downstream systems ?

**Review by asatwal — COMMENTED** [review]
Just one question below.

File: GetIntoTeachingApi/Models/SchoolsExperience/SchoolsExperienceSignUp.cs (hunk 1 above)
Comment (martyn-w) [reply GetIntoTeachingApi/Models/SchoolsExperience/SchoolsExperienceSignUp.cs:171]: nope, they are intentionally optional as older clients will not send the new degree fields. Missing these values means that a qualification record will not be created in the CRM (as is currently the case), but shouldn't break anything downstream.

**Review by asatwal — APPROVED** [review]
👍 LGTM

**Pull request merged**